### DISCUSSION TITLE
Please add binary support to fetch API

### DISCUSSION THREAD

**Discussion by stackrainbow** [zepp-health/zeppos-forum#156] [Developer Feedback]
If I try to retrieve a PNG using the fetch API on side service, the PNG is always corrupted. This is because it is forcibly converted to utf-8.

Use case: Sending the PNG from side service to my remote control media APP which would show now playing media image.

**Comment by stackrainbow** [comment]
If anyone else has this problem, I made a workaround that involves the use of a proxy server.
The proxy server converts a binary response of the original URL and transforms it to base64. Of course this means that the proxy server needs to be extra on-device or in cloud. It is also not very efficent.

So then: app-side -> proxy server fetches > real url -> converts to base64 -> app-side decode to binary -> btle rpc it back to watch side

**Reply by htoooth** [reply]
@stackrainbow 

现在表盘不支持 side-service。应用可以 side-service。fetch 接口很快就会支持二进制功能。但是图片需要转换成我们的格式才可以。

The dial currently does not support side-service, but the application can use side-service. The fetch interface will soon support binary functionality, but images need to be converted to our format before they can be used.

**Comment by vaas1993** [comment]
My first instinct is to try converting the image to base64

**Comment by MyoungXUE** [comment]
Dear @stackrainbow , 

Hi, thank you for your feedback and questions on developing Mini Programs for Zepp OS. Due to hardware limitations of our devices, we apply certain processing to image resources during the Mini Program build process. Only images in the assets directory can be displayed properly on our devices currently. PNG images obtained from network sources cannot be displayed directly on our devices at this time. 
 
We understand the need to support more image formats and network images in Mini Programs. This functionality will be enabled in Zepp OS 3.0. We appreciate your patience and understanding. Please continue to provide us with your valuable feedback. It will help us to build an open platform that serves the needs of our developer community.

Thank you!
Zepp OS Team
